test/aws/osml/model_runner/api/test_image_request.py

```diff
@@ -2,53 +2,226 @@
+
+        # Check S3 Sink creation from outputs
+        sinks = SinkFactory.outputs_to_sinks(ir.outputs)
+        s3_sink: Sink = sinks[0]
+        self.assertEqual(s3_sink.name(), "S3")
+        self.assertEqual(getattr(s3_sink, "bucket"), "test-bucket")
+        self.assertEqual(getattr(s3_sink, "prefix"), "images/outputs")
 
-        ir = ImageRequest(
-            self.sample_request_data,
-            job_id=None,
+    def test_image_request_invalid_sink(self):
+        """
+        Test ImageRequest creation with an invalid sink type.
+        """
+        request = ImageRequest.from_external_message(
+            {
+                "jobName": "test-job-name",
+                "jobId": "test-job-id",
+                "imageUrls": ["test-image-url"],
+                "outputs": [{"type": "SQS", "queue": "FakeQueue"}],
+                "imageProcessor": {"name": "test-model", "type": "SM_ENDPOINT"},
+                "imageProcessorTileSize": 1024,
+                "imageProcessorTileOverlap": 50,
+            }
+        )
+
+        # Should fail with an invalid sync type provided
+        self.assertFalse(request.is_valid())
+
+    def test_image_request_invalid_image_path(self):
+        """
+        Test validation of an invalid S3 image path.
+        """
+        s3_client = boto3.client("s3", config=BotoConfig.default)
+        s3_client_stub = Stubber(s3_client)
+        s3_client_stub.activate()
+
+        image_path = "s3://test-results-bucket/test/data/small.ntf"
+
+        s3_client_stub.add_client_error(
+            "head_object",
+            service_error_code="404",
+            service_message="Not Found",
+            expected_params={"Bucket": image_path},
         )
 
-        assert not ir.is_valid()
+        with self.assertRaises(InvalidS3ObjectException):
+            validate_image_path(image_path, None)
+
+        s3_client_stub.deactivate()
 
     @staticmethod
     def build_request_data():
+        """
+        Helper method to build sample request data for tests.
+        """
         return {
-            "job_id": "test-job",
+            "job_id": "test-job-id",
             "image_id": "test-image-id",
             "image_url": "test-image-url",
             "image_read_role": "arn:aws:iam::012345678910:role/OversightMLS3ReadOnly",
-            "output_bucket": "unit-test",
-            "output_prefix": "region-request",
-            "tile_size": (10, 10),
-            "tile_overlap": (1, 1),
+            "outputs": [
+                {"type": "S3", "bucket": "test-bucket", "prefix": "test-bucket-prefix"},
+                {"type": "Kinesis", "stream": "test-stream", "batchSize": 1000},
+            ],
+            "tile_size": (1024, 1024),
+            "tile_overlap": (50, 50),
             "tile_format": "NITF",
             "model_name": "test-model-name",
-            "model_invoke_mode": "SM_ENDPOINT",
+            "model_invoke_mode": ModelInvokeMode.SM_ENDPOINT,
```

Comment:
If this is meant to simulate a user provided ImageRequest, they would be passing json in the message.  Shouldn't this still be the string value to check that our code properly parses the string into the appropriate enum?

Reply:
Oh this doesn't build an external message it builds an ImageRequest object - so we want the typing here.